### ISSUE TITLE
Tell apart group from users in ACE SID

### ISSUE THREAD

**Issue by freddyMorera** [hierynomus/smbj#377]
I'm trying to use both, smbj and smbj-rpc to retrieve files and their ACE information.

So far I've been able to retrieve the file ACEs, to the point of getting the SID, and then using smbj-rpc to resolve the SIDs, but I haven't been able to check if the ACE is for a user or a group. I only get things like "SYSTEM", "Administrators", "john.doe" with no distinction between them

In the source code I saw a SidType enum with the information I need, but it doesn't seem to be used anywhere.

Is it possible to know if a SID, for example S-1-5-32-544 belongs to a user or a group?

**Comment by hierynomus** [comment]
@pepijnve Do you know?

**Comment by pepijnve** [comment]
It's in the lookup response. See [LsarLookupSids](https://msdn.microsoft.com/en-us/library/cc234488.aspx). That returns a list of [LSAPR_TRANSLATED_NAME](https://msdn.microsoft.com/en-us/library/cc234461.aspx) which contains a [SID_NAME_USE](https://msdn.microsoft.com/en-us/library/cc234454.aspx) field.

**Comment by pepijnve** [comment]
Also accessible in smbj-rpc. See https://github.com/rapid7/smbj-rpc/blob/master/src/main/java/com/rapid7/client/dcerpc/mslsad/objects/LSAPRTranslatedName.java#L106.

**Comment by freddyMorera** [comment]
I forgot to mention my final solution to this issue, thanks to @pepijnve  and @hierynomus  for pointing me to the right direction.

The main issue I had is that the smb-rpc LocalSecurityAuthorityService doesn't expose the LSAPRTranslatedName during the response, it only returns an array of strings, so I extended it and added a method that exposes it. With that approach I was able to retrieve all the information I was looking for.
